### PR TITLE
Add TypeScript, JSX and TSX language support

### PR DESCRIPTION
**Pull request by espenja**
## Praise
I think this is a really cool app, and would love to see it succeed further! ❤️

## What is this PR?
This pull request adds language support and formatting of TypeScript, JSX, TSX.
Also added is autodetection for TypeScript.
JSX and TSX does not seem to be supported by guesslang-js, so autodetection for these are missing.

TypeScript:

image

TSX:

image

JSX:

image

## Tasks
- [x] Add language support for TypeScript, JSX, TSX
- [x] Add formatting support for TypeScript, JSX, TSX
- [x] Add language autodetection for TypeScript

### CODE REVIEW

File: src/editor/languages.js [hunk 1]
```diff
@@ -27,6 +27,9 @@ export const LANGUAGES = [
     new Language("text", "Plain Text", null, null),
     new Language("math", "Math", null, null),
     new Language("javascript", "JavaScript", javascriptLanguage.parser, "js", true),
+    new Language("typescript", "TypeScript", typescriptLanguage.parser, "ts", true),
+    new Language("jsx", "JSX", jsxLanguage.parser, "jsx", true),
+    new Language("tsx", "TSX", tsxLanguage.parser, "tsx", true),
```
Comment (heyman) [inline src/editor/languages.js:32]: I think the fourth argument can be `null` for JSX and TSX, since guesslang doesn't support them.

**Review by heyman — COMMENTED** [review]
Nice!

If you rebase against main Heynote now have a few end-to-end tests as well that should run.

File: src/editor/languages.js (hunk 1 above)
Comment (espenja) [reply src/editor/languages.js:32]: Oh, I thought the last argument represented if the language supports formatting or not?

Comment (espenja) [reply src/editor/languages.js:32]: Wait, my bad, I misinterpreted what you said. Gotcha! I'll do that right now.

**Comment by espenja** [comment]
Three tests are failing after merging in main.
Should this be expected, or have some of my changes broken anything?
image

**Comment by heyman** [comment]
Will take a look! I've just added the tests and I might have messed something up.

**Comment by heyman** [comment]
The tests were broken (for some reason they worked when running in the playwright UI locally). Should be fixed now though!

**Comment by heyman** [comment]
Thanks!

**Pull request merged**